File: algorithmic-toolbox/src/week1/java/LeastCommonMultiple.java
package week1.java;
import java.util.Scanner;

class LeastCommonMultiple {

    public static void main(final String[] args) {
        try (final Scanner in = new Scanner(System.in)) {
            System.out.println(lcm(in.nextLong(), in.nextLong()));
        }
    }

    private static long lcm(final long a, final long b) {
        return (a * b) / gcd(a, b);
    }

    private static long gcd(final long a, final long b) {
        return (b == 0) ? a : gcd(b, a % b);
    }
}

File: algorithmic-toolbox/src/week1/scala/LeastCommonMultiple.scala
object LeastCommonMultiple {
  def main(args: Array[String]) = {
    val in = new java.util.Scanner(System.in)
    println(lcm(in.nextLong, in.nextLong))
  }

  def lcm(a: Long, b: Long): Long = (a*b)/gcd(a, b)
  def gcd(a: Long, b: Long): Long = if (b == 0) a else gcd(b, a % b)
}